Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -21,53 +21,91 @@ import struct CDispatch.dispatch_fd_t
 
 /// A connection between a message handler (e.g. language server) in the same process as the connection object and a remote message handler (e.g. language client) that may run in another process using JSON RPC messages sent over a pair of in/out file descriptors.
 ///
-/// For example, inside a language server, the `JSONRPCConnection` takes the language service implemenation as its `receiveHandler` and itself provides the client connection for sending notifications and callbacks.
-public final class JSONRPCConnection {
+/// For example, inside a language server, the `JSONRPCConnection` takes the language service implementation as its `receiveHandler` and itself provides the client connection for sending notifications and callbacks.
+public final class JSONRPCConnection: Connection {
 
   /// A name of the endpoint for this connection, used for logging, e.g. `clangd`.
-  let name: String
+  private let name: String
 
-  var receiveHandler: MessageHandler? = nil
+  /// The message handler that handles requests and notifications sent through this connection.
+  ///
+  /// Access to this must be be guaranteed to be sequential to avoid data races. Currently, all access are
```

Comment:
`serial` or `synchronized` would make more sense to me than `sequential` here. Same with the other places